Parallelize package validate All checks for faster execution (#15242)
Azure/azure-sdk-tools: tools/azsdk-cli/Azure.Sdk.Tools.Cli/CHANGELOG.md

### Breaking Changes

> - `azsdk package validate` with check type `All` no longer accepts `--fix`. Run a specific check type (for example, `Cspell`, `Snippets`, or `Format`) with `--fix` instead.

Author: ```suggestion
- `azsdk package validate` with check type `All` raises an error when `--fix` is True. Run a specific check type (for example, `Cspell`, `Snippets`, or `Format`) with `--fix` instead.
```

### Bugs Fixed

### Other Changes

> - `azsdk package validate` with check type `All` now runs read-only checks in parallel, significantly reducing wall time on large packages. Potentially-mutating checks (`Snippets`, `Format`) are serialized after the read-only phase to avoid file-write races.

Author: ```suggestion
- `azsdk package validate` with check type `All` now runs read-only checks in parallel, significantly reducing wait time on large packages. Potentially-mutating checks (`Snippets`, `Format`) are serialized after the read-only phase to avoid file-write races.
```
